### PR TITLE
[Synapse] `az synapse role assignment list`: Fix showing only 100 results

### PR DESCRIPTION
**Pull request by wenqiChen0014**
**Description**
Fix the problem that the list command only displays 100 lines

**Testing Guide**

**History Notes**

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

**Review by jiasli — APPROVED** [review]
Approve for core.

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py [hunk 1]
```diff
@@ -33,8 +34,16 @@ def _list_role_assignments(cmd, workspace_name, role=None, assignee=None, scope=
...
-    role_assignments = client.list_role_assignments(role_id, object_id, scope).value
-    return role_assignments
+    token = ""
+    result = []
+    while True:
+        request = client.list_role_assignments(role_id, object_id, scope, continuation_token_parameter=token,
+                                               cls=cust_help.get_deserialized_and_headers)
+        token = request[1]['x-ms-continuation']
+        result += request[0].value
+        if (token == "") or token is None:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:44]: You can simplify it to `if not token`:

**Comment by jsntcy** [comment]
Overall LGTM with a small comment.

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py (hunk 1 above)
Comment (wenqiChen0014) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:44]: OK, I'll try

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**